cmd/cdi-importer/importer.go

```diff
@@ -42,9 +42,19 @@ func main() {
 	ep, _ := util.ParseEnvVar(common.ImporterEndpoint, false)
 	acc, _ := util.ParseEnvVar(common.ImporterAccessKeyID, false)
 	sec, _ := util.ParseEnvVar(common.ImporterSecretKey, false)
+	source, _ := util.ParseEnvVar(common.ImporterSource, false)
+	contentType, _ := util.ParseEnvVar(common.ImporterContentType, false)
 
 	glog.V(1).Infoln("begin import process")
-	err = importer.CopyImage(common.ImporterWritePath, ep, acc, sec)
+	dso := &importer.DataStreamOptions{
+		common.ImporterWritePath,
+		ep,
+		acc,
+		sec,
+		source,
+		contentType,
+	}
+	err = importer.CopyImage(dso)
```

Comment:
So for the blank image feature, I'm using the image package where I will have a method to create the image in the qemu.go file. I do not need to use the DataStream file of course. Anyway... The  only parameters I will need to create the image are the imageSize (which will be another env var) and the common.ImporterWritePath. I do not want to pass the entire dso struct. Can we move the ImporterWritePath out of it?

Reply:
Right, but you have a method signature something like
```go
func BlankImage(size int64) error {
  ...
}
```
Then you can check
```go
if source == SourceNone && contentType == ContentTypeKubevirt {
   BlankImage(pvcRequestSize)
}
```

Reply:
The question is how to pass to BlankImage() the common.ImporterWritePath argument without passing the whole dso object...

Reply:
oh I see. what you saying is that I still need the  source and contentType...

Reply:
maybe change the structure name as it's not relevant to datastream only?

Reply:
No the check is done in the importer main, based on the values of the env variables you either call blank image or go through the normal path. Since common.ImporterWritePath is a global const, there is no need to pass it around. The only reason its part of the struct is so that the tests can pass in something different. If you need it for the tests, you can add a parameter to the BlankImage function.

Reply:
No the structure is for the DataStream info only. Blank image is a very special case that goes outside the normal flow completely.

Reply:
okay.